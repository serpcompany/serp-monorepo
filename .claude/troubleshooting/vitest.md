https://vitest.dev/guide/common-errors.html