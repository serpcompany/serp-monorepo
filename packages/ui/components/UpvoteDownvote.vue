<template>
  <div class="flex flex-col items-center justify-center">
    <div class="mb-2 flex flex-col items-center">
      <button class="p-1 text-neutral-500" type="button">
        <svg
          xmlns="http://www.w3.org/2000/svg"
          class="h-5 w-5"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
        >
          <path
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            d="M5 15l7-7 7 7"
          />
        </svg>
      </button>
      <span class="mx-1 text-sm font-bold">#</span>
      <button class="p-1 text-neutral-500" type="button">
        <svg
          xmlns="http://www.w3.org/2000/svg"
          class="h-5 w-5"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
        >
          <path
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            d="M19 9l-7 7-7-7"
          />
        </svg>
      </button>
    </div>
  </div>
</template>

<script setup lang="ts"></script>
