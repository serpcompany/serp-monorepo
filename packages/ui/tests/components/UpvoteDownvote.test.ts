import { describe, expect, it } from 'vitest';
import UpvoteDownvote from '../../components/UpvoteDownvote.vue';
import ComponentRender from '../componentRender';

describe('UpvoteDownvote', () => {
    it('renders correctly', async () => {
        const html = await ComponentRender('UpvoteDownvote', {}, UpvoteDownvote)
        expect(html).toMatchSnapshot()
    })
})