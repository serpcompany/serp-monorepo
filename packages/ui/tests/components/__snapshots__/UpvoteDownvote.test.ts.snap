// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UpvoteDownvote > renders correctly 1`] = `
"<div class="flex flex-col items-center justify-center">
  <div class="mb-2 flex flex-col items-center"><button class="p-1 text-neutral-500" type="button"><svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5" fill="none" viewBox="0 0 24 24" stroke="currentColor">
        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M5 15l7-7 7 7"></path>
      </svg></button><span class="mx-1 text-sm font-bold">#</span><button class="p-1 text-neutral-500" type="button"><svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5" fill="none" viewBox="0 0 24 24" stroke="currentColor">
        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M19 9l-7 7-7-7"></path>
      </svg></button></div>
</div>"
`;
